### ISSUE TITLE
Slow and extremely high memory usage (>10GB)

### ISSUE THREAD

**Issue by ychang4** [mltframework/shotcut#880]
Just download and install the shotcut-win64-200412.exe and use on a very small clip (50MB .mkv, 6 min, 1080 screen captured from OBS) to add subtitle. The computer is the Lenovo T490, 24GB memory, 1 TB NVME drive and Intel i7 CPU, 4 cores/8 threads with Win10. CPU usage is about 12~20% throughout.

New to this, not sure what's the right way to do it properly, so I break up the clip into ~ 40 fragments and half of the fragments with a Text_simple filter for the subtitle. The more fragments it has the slower it became and higher memory usage will be. Eventually it consumes > 13GB. It is extremely slow with the high amount of memory usage(like over 3 GB) that every action(like select a segment in the timeline, add a new text filter, save project, export video) will take over an hour or 2 to respond. Also noticed the list of filters failed to scroll properly.  Arrow key does not work. It does scroll when using the scroll bar, but all entries are from the filters in the 1st page.

The workaround is to save/close it when it's over 4GB memory and restart/reload the app/project. After the reload, the memory usage is about 3.5GB. Any action(break up fragment, add new filter, copy/paste filter) performs after that will increase the amount of memory usage a lot, in GB. Perform couple actions and then save/restart the app, process repeat. Shouldn't be too difficult to reproduce the problem, but let me know if you fail to reproduce.

**Comment by ddennedy** [comment]
I did not reproduce it. I added a 10 minute 1920x1080 clip to the timeline, split 40 times, and added Text: Simple filter to each timeline clip. At the end of that:
![image](https://user-images.githubusercontent.com/1146683/85079333-6f9ee700-b17b-11ea-9a06-e49f4d80bfcc.png)

Then, I clicked each clip to select it, and then played it from beginning to end. At the end of that:
![image](https://user-images.githubusercontent.com/1146683/85079385-8a715b80-b17b-11ea-9d5a-1f3a43185334.png)

I do not deny that it gets slower as you add more clips, filters, and keyframes but that is just the way it is and not going to be able to fix. Video editing is inherently heavy and especially at HD or higher. That is why our *minimum* [system requirements](https://shotcut.org/FAQ/#what-are-the-minimum-system-requirements) dictates 8 GB for HD.

Also, I do not accept bugs that are like "it should use less memory" or "it should be faster" just as I do not accept a bug that says "there should be no bugs." Of course, these are overall goals, but not specific enough and actionable. Rather to improve performance and reduce memory, we added preview scaling in version 20.02 to be used in conjunction with [proxy editing](https://forum.shotcut.org/t/proxy-editing/18517) in the next version 20.06 (beta available in releases).

Lastly, I strongly discourage you from splitting clips just to add filters at certain times. For text, I recommend you add a video track and then put text clips (Open Other > Text) on this new track. When I do not split the clip I am using for test and add 42 text clips on V2 and play it from beginning to end:

![image](https://user-images.githubusercontent.com/1146683/85080108-38313a00-b17d-11ea-8f48-2bd3acbee166.png)

To control the timing of other filters, you can use [filter trimming](https://forum.shotcut.org/t/trimming-filters/13212).

**Comment by ddennedy** [comment]
After repeating all of the steps of the first test above (splitting and adding text filter to each, selecting, each, playing begin to end) using proxy and preview scaling at 540p in the v20.06 beta2:

![image](https://user-images.githubusercontent.com/1146683/85081308-7845ec00-b180-11ea-9a4c-49ae2d338859.png)

**Comment by ychang4** [comment]
Thanks for your quick response! I wasn't aware that there's a newer v20.06 beta2, so I downloaded it and was able to recreate the problem with the latest version.

The problem wasn't so much about the 40 segments and the filters, but it was the problem with copy/past of the filters. Since I'm new to this and just want to get some subtitle done, the steps I did might have contribute to the problem I encountered. Here're the exact steps I did.

1. Drag/drop a clip to the timeline
2. Add a "Text: Simple" filter to the clip
3. Play 3 seconds, pause at the start of my next subtitle
4. Split the clip at the head pointer
5. Copy/Paste the filters
6. Modify the top of the filter to the new subtitle
7. Repeat 3~6

My problem is with the **step 5**, where the copy/paste of the filters actually **double** the amount of filters to the previous segment. The amount of filters become exponential growth. After created 15 segments, the copy/paste filters actually will generate 2^14 for the 15th segment and the total of filters will be 65535. The next segment will have 65536 filters and the total would be 131071 filters. The memory consumption after copy/paste at the step 5 of the 16th segment is over 6 GB.

![image](https://user-images.githubusercontent.com/34361707/85095281-c3262a80-b1a5-11ea-8435-97e3abf4b8ac.png)

To summarize my experience as a new user, I need to spend more time to learn the proper way of adding a subtitle in shotcut. However I think there're 3 changes that can help me to avoid this problem.
1. On step 4, when the clip is split, has an option to carry the filters to the next clip (default disable)
    -> I was under impression that the filters are shared globally, since they are duplicated/carried to the new segment. I didn't realize they are unique to each segment and uses copy/paste for the subtitle I made earlier(with a particular format/location I want). 

2. On step 5, when the filters is copied/pasted, it copies the whole set, instead of just the one highlighted. If it only copy/paste the highlighted one, I wouldn't run into this problem and a new option would be nice.

3. Add another filter named specifically as the subtitle. When that is added, it will create a new "subtitle" track automatically to deposit that filter without user intervention. (or as an option). 

In addition, the arrow keys do not work on the filter list. Mouse scroll works on the scroll bar, just not as precise. Thanks for sharing your work, cheers!

**Comment by ddennedy** [comment]
What? I used copy and paste filters too, but I only copied once. You are adding too many filters; that is on you. Do not continually copying and paste multiple. And delete redundant or unnecessary fillers after pasting.

**Comment by ychang4** [comment]
Sorry for the noise and appreciated for your prompt support!

**Comment by ddennedy** [comment]
A few more points:

1. Split DOES keep the filters that was on the clip being split. 
2. Paste does not double the number of filters... unless you double-click that button. You are not supposed to double-click it. Most buttons in most apps are not double-click.
3. The arrow keys do scroll the filter list for me.
4. Many people have asked for a way to select which filters are copied, but there is no multi-selection here. This is low priority, and we have a lot to do with very few people to help only part-time.

**Comment by ychang4** [comment]
> 2. Paste does not double the number of filters... unless you double-click that button. You are not supposed to double-click it. Most buttons in most apps are not double-click.
-> I did not double-click. After retrying few more times, this is what I observed. If I only click the copy once, then just continuous pasting it throughout the project, then I only see one added for every click to the paste button. However if I click the copy button again(should not considered as double-click) some time later at some point, then the paste button will paste double amount of the current filters. Here're my steps

1. Add a filter to the 1st segment                -> Total 1 filter
2. Split the segment, copy/paste the filters -> will add a new filter, total 2 filters
3. Split the segment, paste the filters          -> will add a new filter, total 3 filters
4. Split the segment and add a new filter from the selection -> Total 4 filters
5. Copy/Paste the filters                               -> will add 4 new filters, total 8 filters

> 3. The arrow keys do scroll the filter list for me.
-> I started from a clean project, do nothing but added 2 text filters. The arrow keys still do not work for me through.

Now that I understand the software better, I'm glad my main problem is resolved and thanks for your additional clarifications.
